src/users/users.service.ts

```diff
@@ -53,8 +59,31 @@ export class UsersService {
     }
   }
 
-  async update(id: number, updateUserDto: PartialUserDto) {
-    return `This action updates a #${id} user`;
+  async update(id: UUID, updateUserDto: PartialUserDto) {
+    try {
+      const user = await this.findOne(id);
+      if (
```

Comment:
Instead of checking for non-wanted values, you can define your update user DTO using the helper class [`PickType`](https://docs.nestjs.com/openapi/mapped-types#pick) from nestjs and only selecting the fields you want to update